source/shared/cpp/AdaptiveCardsSharedModel/AdaptiveCardsSharedModelUnitTest/TimeAndDateFucntionTest.cpp

```diff
@@ -0,0 +1,207 @@
+#include "stdafx.h"
+#include "CppUnitTest.h"
+#include "TextBlock.h"
+#include <time.h>
+
+using namespace Microsoft::VisualStudio::CppUnitTestFramework;
+using namespace AdaptiveCards;
+using namespace std;
+
+namespace AdaptiveCardsSharedModelUnitTest
+{		
+	TEST_CLASS(UnitTest1)
+	{
+	public:
+		TEST_METHOD(ISO8601TransformTest)
+		{
+			TextBlock* blck = new TextBlock();
+			Assert::IsNotNull(blck);
+            blck->SetText("{{DATE(2017-02-13T20:46:30Z, Short)}}");
+			Assert::AreEqual<string>("02/13/2017",  blck->GetText());
+			delete blck;
+		}
+		TEST_METHOD(ISO8601TransformTestLong)
+		{
+			TextBlock* blck = new TextBlock();
+			Assert::IsNotNull(blck);
+			string testString = "{{DATE(2017-02-13T20:46:30Z, Long)}}";
+			blck->SetText(testString);
+			Assert::AreEqual<string>("Monday, February 13, 2017", blck->GetText());
+			delete blck;
+		}
+		TEST_METHOD(ISO8601TransformTestDefault)
+		{
+			TextBlock* blck = new TextBlock();
+			Assert::IsNotNull(blck);
+			string testString = "{{DATE(2017-02-13T20:46:30Z)}}";
+			blck->SetText(testString);
+			Assert::AreEqual<string>("02/13/2017", blck->GetText());
+			delete blck;
+		}
+
+		TEST_METHOD(ISO8601TransformBadInputTest)
+		{
+			TextBlock* blck = new TextBlock();
+			Assert::IsNotNull(blck);
+			string testString = "{a{DATE(2017-02-13T20:46:30Z, Short)}}";
+			blck->SetText(testString);
+			Assert::AreEqual<string>("{a{DATE(2017-02-13T20:46:30Z, Short)}}", blck->GetText());
+			delete blck;
+		}
+
+		TEST_METHOD(ISO8601TransformBadInputTest2)
```

Comment:
Yay! Tests! could you name these in such a way that they specifically call out what is wrong with this input?

Reply:
good point

---
In reply to: [148399471](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r148399471) [](ancestors = 148399471)